extensions/amp-animation/0.1/web-animations.js

```diff
@@ -0,0 +1,363 @@
+   * }} context
+   * @param {boolean} validate
+   */
+  constructor(win, context, validate) {
+    super();
+    /** @const */
+    this.win = win;
+
+    /** @const @private */
+    this.context_ = context;
+
+    /** @const @private */
+    this.validate_ = validate;
+
+    /** @private {!Array<!InternalWebAnimationRequestDef>} */
+    this.requests_ = [];
+
+    /** @private {!WebAnimationTimingDef} */
+    this.timing_ = {
+      duration: 0,
+      delay: 0,
+      endDelay: 0,
+      iterations: 1,
+      iterationStart: 0,
+      easing: 'linear',
+      direction: WebAnimationTimingDirection.NORMAL,
+      fill: WebAnimationTimingFill.NONE,
+    };
+
+    /** @private {!Array<!Element> } */
+    this.targets_ = [];
+
+    /** @private {!Array<!CSSStyleDeclaration>} */
+    this.computedStyleCache_ = [];
+  }
+
+  /** @return {!WebAnimationRunner} */
+  createRunner() {
+    return new WebAnimationRunner(this.requests_);
+  }
+
+  /** @override */
+  onMultiAnimation(spec) {
+    const newTiming = this.mergeTiming_(spec, this.timing_);
+    this.withTiming_(newTiming, () => {
+      this.scan(spec.animations);
+    });
+  }
+
+  /** @override */
+  onKeyframeAnimation(spec) {
+    const timing = this.mergeTiming_(spec, this.timing_);
+    const target = this.resolveTarget_(spec.target);
+    /** @type {!WebKeyframesDef} */
+    let keyframes;
+
+    if (isObject(spec.keyframes)) {
+      // Property -> keyframes form.
+      const object = /** {!Object<string, *>} */ (spec.keyframes);
+      keyframes = {};
+      for (const prop in object) {
+        const value = object[prop];
+        let preparedValue;
+        if (!isArray(value) || value.length == 1) {
+          // Missing "from" value. Measure and add.
+          const fromValue = this.measure_(target, prop);
+          preparedValue = [fromValue, clone(isArray(value) ? value[0] : value)];
+        } else {
+          preparedValue = clone(value);
+        }
+        keyframes[prop] = preparedValue;
+      }
+    } else if (isArray(spec.keyframes) && spec.keyframes.length > 0) {
+      // Keyframes -> property form.
+      const array = /** {!Array<!Object<string, *>>} */ (spec.keyframes);
+      keyframes = [];
+      // Fill-in partial key-frames.
+      // See https://github.com/web-animations/web-animations-js/issues/14
+      const addStartFrame = array.length == 1 || array[0].offset > 0;
+      const startFrame = addStartFrame ? {} : clone(array[0]);
+      keyframes.push(startFrame);
+      const start = addStartFrame ? 0 : 1;
+      for (let i = start; i < array.length; i++) {
+        const frame = array[i];
+        for (const prop in frame) {
+          if (!startFrame[prop]) {
+            // Missing "from" value. Measure and add to start frame.
+            startFrame[prop] = this.measure_(target, prop);
+          }
+        }
+        keyframes.push(clone(frame));
+      }
+    } else {
+      // No a known form of keyframes spec.
```

Comment:
remove 'a'

Took a bit of reading to get what `onKeyframeAnimation` does, but it is fairly clear what it does after. Some summary comments on `onKeyframeAnimation` may help future readers.